src/main/scala/is/hail/methods/LinearMixedRegressionDist.scala

```diff
@@ -0,0 +1,216 @@
+package is.hail.methods
+
+import breeze.linalg._
+import is.hail.annotations._
+import is.hail.distributedmatrix.DistributedMatrix
+import is.hail.expr._
+import is.hail.stats._
+import is.hail.utils._
+import is.hail.variant.VariantDataset
+import org.apache.spark.mllib.linalg.{DenseMatrix => SparkDenseMatrix}
+import org.apache.spark.mllib.linalg.distributed.{BlockMatrix, IndexedRow}
+import is.hail.distributedmatrix.DistributedMatrix.implicits._
+
+object LinearMixedRegressionDist {
+  val dm = DistributedMatrix[BlockMatrix]
+
+  import dm.ops._
+
+  def applyEigenDist(
+    vds: VariantDataset,
+    eigenDist: EigendecompositionDist,
+    yExpr: String,
+    covExpr: Array[String],
+    useML: Boolean,
+    rootGA: String,
+    rootVA: String,
+    runAssoc: Boolean,
+    optDelta: Option[Double],
+    sparsityThreshold: Double,
+    useDosages: Boolean): VariantDataset = {
+
+    require(vds.wasSplit)
+
+    val pathVA = Parser.parseAnnotationRoot(rootVA, Annotation.VARIANT_HEAD)
+    Parser.validateAnnotationRoot(rootGA, Annotation.GLOBAL_HEAD)
+
+    val (y, cov, completeSamples) = RegressionUtils.getPhenoCovCompleteSamples(vds, yExpr, covExpr)
+    val C = cov
+    val completeSamplesSet = completeSamples.toSet
+    val sampleMask = vds.sampleIds.map(completeSamplesSet).toArray
+    val completeSampleIndex = (0 until vds.nSamples)
+      .filter(i => completeSamplesSet(vds.sampleIds(i)))
+      .toArray
+
+    optDelta.foreach(delta =>
+      if (delta <= 0d)
+        fatal(s"delta must be positive, got ${ delta }"))
+
+    val covNames = "intercept" +: covExpr
+
+    val n = y.length
+    val c = C.cols
+    val d = n - c - 1
+
+    if (d < 1)
+      fatal(s"lmmreg: $n samples and $c ${ plural(c, "covariate") } including intercept implies $d degrees of freedom.")
+
+    info(s"lmmreg: Running lmmreg on $n samples with $c sample ${ plural(c, "covariate") } including intercept...")
+
+    optDelta match {
+      case Some(del) => info(s"lmmreg: Delta of $del specified by user")
+      case None => info(s"lmmreg: Estimating delta using ${ if (useML) "ML" else "REML" }... ")
+    }
+
+    // FIXME: need filtering on eigenDist
+    val EigendecompositionDist(_, rowIds, evects, evals) = eigenDist
+
+    if (!completeSamples.sameElements(rowIds))
+      fatal("Bad stuff")
+
+    val Ut = evects.transpose
```

Comment:
Confirm this with Dan, but I think that that just as using `multiply` is worse than using `*` on BlockMatrices now,  I think using `transpose` is worse than using `t` as imported from `DistributedMatrix`.